### PR TITLE
GEODE-1617: Regions can be created with a variety of characters that …

### PR DESCRIPTION
**Pull request by kjduling**
…are unsupported

Added comprehensive test to validate region names

### CODE REVIEW

**Comment by kirklund** [comment]
+1

**Comment by kirklund** [comment]
I reworded your commit message:

```
GEODE-1617: add test for region names

Added comprehensive test to validate region names

This closes #285
```

File: geode-core/src/test/java/org/apache/geode/cache/RegionNameValidationJUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
+package org.apache.geode.cache;
```
Comment (kirklund) [inline geode-core/src/test/java/org/apache/geode/cache/RegionNameValidationJUnitTest.java]: Missing the Apache license header on this file.

File: geode-core/src/test/java/org/apache/geode/cache/RegionNameValidationJUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+  }
+
+  @Test
+  public void testExternalRegionNames() {
+    InternalRegionArguments ira = new InternalRegionArguments();
+    ira.setInternalRegion(false);
+    validateCharacters(ira);
+    try {
+      LocalRegion.validateRegionName("__InvalidInternalRegionName", ira);
```
Comment (kirklund) [inline geode-core/src/test/java/org/apache/geode/cache/RegionNameValidationJUnitTest.java]: Aren't you missing a fail() stmt here?

File: geode-core/src/test/java/org/apache/geode/cache/RegionNameValidationJUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  private void validateCharacters(InternalRegionArguments ira) {
+    for (int x = 0; x < Character.MAX_VALUE; x++) {
+      String name = (char) x + REGION_NAME;
+      Matcher matcher = NAME_PATTERN.matcher(name);
+      if (matcher.matches()) {
+        LocalRegion.validateRegionName(name, ira);
+      } else {
+        try {
+          LocalRegion.validateRegionName(name, ira);
+          fail("Should have received an IllegalArgumentException");
```
Comment (kirklund) [inline geode-core/src/test/java/org/apache/geode/cache/RegionNameValidationJUnitTest.java]: Recommend adding which character to the message so we get instant info about which character should have been illegal but was allowed.

File: geode-core/src/test/java/org/apache/geode/cache/RegionNameValidationJUnitTest.java (hunk 1 above)
Comment (kjduling) [inline geode-core/src/test/java/org/apache/geode/cache/RegionNameValidationJUnitTest.java]: Hmm, my local build didn't catch that and should have.  Fixed.

File: geode-core/src/test/java/org/apache/geode/cache/RegionNameValidationJUnitTest.java (hunk 2 above)
Comment (kjduling) [inline geode-core/src/test/java/org/apache/geode/cache/RegionNameValidationJUnitTest.java]: Yes, fixed.

File: geode-core/src/test/java/org/apache/geode/cache/RegionNameValidationJUnitTest.java (hunk 3 above)
Comment (kjduling) [inline geode-core/src/test/java/org/apache/geode/cache/RegionNameValidationJUnitTest.java]: I added printing the offending character and the numeric value of it.

**Comment by kirklund** [comment]
Pushed to develop. Thanks!

**Pull request closed without merging**